### PR TITLE
Integrate CSS with bundler

### PR DESCRIPTION
**Pull request by zackradisic**
### What does this PR do?

Parses, prints, minifies CSS in the bundler in canary or debug builds.

### CODE REVIEW

**Comment by robobun** [comment]
@zackradisic, your commit c7912b6d3e4ece39467cf680497820c1e6ca998e is building: [`#4185`](https://buildkite.com/bun/bun/builds/4185)

File: src/css/printer.zig [hunk 1]
```diff
@@ -72,6 +72,15 @@ const Browsers = css.targets.Browsers;
 /// `Printer` also includes helper functions that assist with writing output
 /// that respects options such as `minify`, and `css_modules`.
 pub fn Printer(comptime Writer: type) type {
+    const getWrittenAmt = struct {
```
Comment (paperclover) [inline src/css/printer.zig]: why not put this as a function within the `struct {` closure. this anonymous struct is not needed.

File: src/bundler/bundle_v2.zig [hunk 2]
```diff
@@ -3115,16 +3115,44 @@ pub const ParseTask = struct {
...
+                        .top_level_await_keyword = Logger.Range.None,
+                        .approximate_newline_count = 0, // TODO: this
+                        .import_records = import_records,
+                        .css = css_ast_heap,
+                        // .allocator = bundler.allocator,
+                        .parts = js_ast.Part.List.fromSlice(bundler.allocator, &.{ .{}, .{} }) catch bun.outOfMemory(),
+                    };
+                }
             },
+            // TODO: css
```
Comment (paperclover) [inline src/bundler/bundle_v2.zig]: stale comment

File: src/bundler/bundle_v2.zig [hunk 3]
```diff
@@ -4788,10 +4817,38 @@ pub const LinkerContext = struct {
             var entry_bits = &this.graph.files.items(.entry_bits)[source_index];
             entry_bits.set(entry_bit);
 
+            const key = try temp_allocator.dupe(u8, entry_bits.bytes(this.graph.entry_points.len));
```
Comment (paperclover) [inline src/bundler/bundle_v2.zig]: unused

File: src/bundler/bundle_v2.zig [hunk 4]
```diff
@@ -7530,6 +8110,249 @@ pub const LinkerContext = struct {
...
+        wg: *sync.WaitGroup,
+    };
+
+    fn prepareCssAstsForChunk(task: *ThreadPoolLib.Task) void {
+        const prepare_css_asts: *const PrepareCssAstTask = @fieldParentPtr("task", task);
+        defer prepare_css_asts.wg.finish();
+        var worker = ThreadPool.Worker.get(@fieldParentPtr("linker", prepare_css_asts.linker));
+        defer worker.unget();
+
+        // const prev_action = if (Environment.isDebug) bun.crash_handler.current_action;
```
Comment (paperclover) [inline src/bundler/bundle_v2.zig]: remove comment or uncomment

**Review by paperclover — CHANGES_REQUESTED** [review]
note: did not review all code closely, but i think there wont be ~~real issues outside of canary where this flag is enabled~~ nvm there are currently test regressions

**Comment by Jarred-Sumner** [comment]
merge conflikcts

File: src/linker.zig [hunk 5]
```diff
@@ -281,6 +281,10 @@ pub const Linker = struct {
                             continue;
                         }
 
+                        if (strings.hasSuffixComptime(import_record.path.text, ".css")) {
```
Comment (Jarred-Sumner) [inline src/linker.zig]: ✂️

**Pull request merged**